devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,215 @@
+				attributeState "both", label: 'ring!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#0e7507"
+			}
+		}
+		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ffffff"
+		}
+
+		main "alarm"
+		details(["alarm", "off", "tamper"])
+	}
+}
+
+private getNumberOfSounds() {
+	def numberOfSounds = [
+            "0003": 8 //Aeotec Doorbell 6
+	]
+	return numberOfSounds[zwaveInfo.prod] ?: 1
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "alarm", value: "off", isStateChange: true, displayed: false)
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if (!childDevices) {
+		addChildren(numberOfSounds)
+	}
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def securedEncapsulatedCommand = cmd.securedEncapsulatedCommand()
+	if (securedEncapsulatedCommand) {
+		zwaveEvent(securedEncapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+
+}
+
+private secure(physicalgraph.zwave.Command cmd) {
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private onOffCmd(value) {
+    secure(zwave.basicV1.basicSet(value: value))
+}
+
+def on() {
+	resetActiveSound()
+	state.lastTriggeredSound = 1
+	onOffCmd(0xFF)
+}
+
+def off() {
+	state.lastTriggeredSound = 1
+	onOffCmd(0x00)
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	secure(zwave.basicV1.basicGet())
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 1..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : "Sound $endpoint",
```

Comment:
Sorry, I haven't been following this PR but just hopped in to check the usage of `addChildDevice` since we've had a lot of problems in this area. I would suggest leaving the `componentLabel` as is but changing the `label` to include the `device.displayName`. You can look at "Zigbee Multi Button" or "Z-Wave Multi Button" for examples. Groovy SmartApps aren't MCD aware so they only show the `label` of each device which is why it's helpful to include the `displayName` of the main component in each child component's `label`.

Reply:
Hi @tpmanley 
So I should leave here "Aeotec Doorbell + endpoint number" - do not do it in the way - sound + endpoint? 

I checked those DTH's which you mentioned and name format is Aeotec Doorbell + endpoint - I assume this is correct. But what do you think about this?

Reply:
I think: `label: "$device.displayName Sound $endpoint"`

Reply:
I'll leave most of the review to him :). I'm just focusing on the MCD parts. You'll also need to set `mcdSync: true` in the description in order for this to truly be an MCD device.